### PR TITLE
test(apm): fix apm test teardown

### PR DESCRIPTION
**Pull request by daprahamian**
APM tests are doing some instrumentation that is not undone
at the end of the test. This would prevent further tests from
completing. This change fixes one of the offending tests, while also
making it easy to do so for future tests.

### CODE REVIEW

File: test/functional/apm_tests.js [hunk 1]
```diff
@@ -125,10 +132,12 @@ describe('APM', function() {
           .then(function(r) {
             expect(r.insertedCount).to.equal(1);
 
-            var listener = require('../..').instrument(function(err) {
+            self.listener = require('../..').instrument(function(err) {
```
Comment (mbroadst) [inline test/functional/apm_tests.js]: I would prefer not to use the magic context for this.  I tend to make a global variable called e.g. `test` or `testContext` to make it very explicit where this stuff lives

Comment (mbroadst) [inline test/functional/apm_tests.js]: I would prefer not to use the magic context for this.  I tend to create a global variable e.g. `test` or `testContext` to make it very explicit where all this stuff lives

File: test/functional/apm_tests.js [hunk 2]
```diff
@@ -10,6 +10,13 @@ describe('APM', function() {
     return setupDatabase(this.configuration);
   });
 
+  afterEach(function() {
+    if (this.listener) {
+      this.listener.uninstrument();
```
Comment (mbroadst) [inline test/functional/apm_tests.js]: What if its already uninstrumented? Presumably some cases here don't fail

Comment (daprahamian) [reply test/functional/apm_tests.js]: I believe i tested this, and calling `uninstrument` multiple times does not die IIRC

**Pull request merged**